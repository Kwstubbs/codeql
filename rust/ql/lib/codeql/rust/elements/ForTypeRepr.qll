// generated by codegen, do not edit
/**
 * This module provides the public class `ForTypeRepr`.
 */

private import internal.ForTypeReprImpl
import codeql.rust.elements.ForBinder
import codeql.rust.elements.TypeRepr

/**
 * A type with a higher-ranked `for` modifier. This is currently not valid Rust syntax (`for<...>` can
 * only be applied to traits to form a `TypeBound`).
 *
 * For example:
 * ```rust
 * fn foo(value: for<'a> usize) {}  // DOESN'T COMPILE
 * //            ^^^^^^^^^^^^^
 * ```
 */
final class ForTypeRepr = Impl::ForTypeRepr;
