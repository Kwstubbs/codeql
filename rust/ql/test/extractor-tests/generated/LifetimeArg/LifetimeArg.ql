// generated by codegen, do not edit
import codeql.rust.elements
import TestUtils

query predicate instances(LifetimeArg x, string hasLifetime__label, string hasLifetime) {
  toBeTested(x) and
  not x.isUnknown() and
  hasLifetime__label = "hasLifetime:" and
  if x.hasLifetime() then hasLifetime = "yes" else hasLifetime = "no"
}

query predicate getLifetime(LifetimeArg x, Lifetime getLifetime) {
  toBeTested(x) and not x.isUnknown() and getLifetime = x.getLifetime()
}
