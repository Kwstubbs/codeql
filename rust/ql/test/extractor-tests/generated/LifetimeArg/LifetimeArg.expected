instances
| gen_lifetime_arg.rs:7:20:7:21 | LifetimeArg | hasLifetime: | yes |
getLifetime
| gen_lifetime_arg.rs:7:20:7:21 | LifetimeArg | gen_lifetime_arg.rs:7:20:7:21 | 'a |
